### ISSUE TITLE
Kaniko build on Google Cloud Build doesn't include --target parameter

### ISSUE THREAD

**Issue by iamarcel** [GoogleContainerTools/skaffold#3399]
### Expected behavior
When scheduling a Kaniko build on GCB, the artifact's Target parameter gets passed as an argument to the Kaniko executor on GCB, just like when building it in-cluster.

### Actual behavior
The Kaniko build spec function [doesn't include code to create a `--target` parameter](https://github.com/GoogleContainerTools/skaffold/blob/42330f2/pkg/skaffold/build/gcb/kaniko.go#L30-L63) (while the respective method for in-cluster builds [does](https://github.com/GoogleContainerTools/skaffold/blob/42330f2/pkg/skaffold/build/cluster/kaniko.go#L129-L131)).

The tests for the GCB Kaniko build spec [don't include the Target parameter](https://github.com/GoogleContainerTools/skaffold/blob/42330f20d0a4a02e9d4386d2002b1b4f2111ecd8/pkg/skaffold/build/gcb/kaniko_test.go#L29) either.

### Information

- Skaffold version: v1.0.1
- Operating system: Fedora 31
- Relevant contents of skaffold.yaml:

```yaml
build:
  googleCloudBuild:
    projectId: my-google-project
  artifacts:
  - image: eu.gcr.io/my-google-project/build
    kaniko:
      target: build
      cache: {}
      buildContext:
        gcsBucket: my-kaniko-bucket
```

The GCB build details shows these arguments:

```
--destination eu.gcr.io/my-google-project/build:f4c58a6 --dockerfile Dockerfile --cache
```

**Comment by balopat** [comment]
Thank you @iamarcel - I agree that this is missing, and you are pointing to the right places. PRs are welcome!

**Comment by hwanjin-jeong** [comment]
I contribute to this issue
Please see that you want to check the resolved code: #3462

**Comment by dgageot** [comment]
Fixed with https://github.com/GoogleContainerTools/skaffold/pull/3462
Thanks a lot @hwanjin-jeong !
